File: hello-proxy-api/src/main/scala/com/example/helloproxy/api/HelloProxyService.scala
/*
 * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
 */

package com.example.helloproxy.api

import akka.NotUsed
import com.lightbend.lagom.scaladsl.api.transport.Method
import com.lightbend.lagom.scaladsl.api.{ Service, ServiceCall }

trait HelloProxyService extends Service {

  def proxyViaHttp(id:String): ServiceCall[NotUsed, String]
  def proxyViaGrpc(id:String): ServiceCall[NotUsed, String]

  override final def descriptor = {
    import Service._

    named("hello-proxy")
      .withCalls(
        restCall(Method.GET, "/proxy/rest-hello/:id", proxyViaHttp _),
        restCall(Method.GET, "/proxy/grpc-hello/:id", proxyViaGrpc _)
      ).withAutoAcl(true)
  }
}

File: grpc-example/grpc-example-java/hello-proxy-api/src/main/java/com/example/helloproxy/api/HelloProxyService.java
package com.example.helloproxy.api;

import akka.NotUsed;
import com.lightbend.lagom.javadsl.api.Descriptor;
import com.lightbend.lagom.javadsl.api.Service;
import com.lightbend.lagom.javadsl.api.ServiceCall;
import com.lightbend.lagom.javadsl.api.transport.Method;

import static com.lightbend.lagom.javadsl.api.Service.*;

public interface HelloProxyService extends Service {

    ServiceCall<NotUsed, String> proxyViaHttp(String id);
    ServiceCall<NotUsed, String> proxyViaGrpc(String id);

    default Descriptor descriptor() {
        return named("hello-proxy")
            .withCalls(
                restCall(Method.GET, "/proxy/rest-hello/:id", this::proxyViaHttp),
                restCall(Method.GET, "/proxy/grpc-hello/:id", this::proxyViaGrpc)
            ).withAutoAcl(true);
    }
}

